src/averageAges.js

```diff
@@ -55,7 +72,28 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person => (
+    people.find(mother => person.mother === mother.name))
+    && (onlyWithSon
+      ? person.sex === 'm'
+      : true)
+  );
+
+  const differenceAges = children.map(child => {
+    const mother = people.find(person => person.name === child.mother);
+
+    const differenceAge = child.born - mother.born;
+
+    return differenceAge;
+  });
+
+  const totalAge = differenceAges.reduce((age, average) => age + average);
```

Comment:
You use reduce in every function. It should be added to the `calculateAverageAge` function.